lib/load/worker.ex

```diff
@@ -43,16 +43,29 @@ defmodule Load.Worker do
   end
 
 
-  def handle_info(:run, %{sim: sim, interval_ms: interval_ms} = state) do
-    state = state
-    |> sim.run()
+  def handle_info(:run, %{sim: sim, interval_ms: interval_ms, latencies: latencies} = state) do
+    start_time = System.monotonic_time
+
+    # TODO: Make this fire and forget to not be affected by the response time.
+    state = sim.run(state)
+
+    latency_ms = (System.monotonic_time - start_time) / 1000000 |> trunc()
+
+    Logger.debug("Request latency: #{latency_ms} ms")
+
+    state = %{state | latencies: [latency_ms] ++ latencies}
     |> Stats.maybe_update()
-    Process.send_after(self(), :run, interval_ms)
+
+    Process.send_after(self(), :run, (interval_ms - latency_ms))
```

Comment:
I would restore the original here, this assumes latency is always smaller than interval. This is dangerous, and a bit unnecessary